File: leetcode/344. 反转字符串.go
package main

// 344. 反转字符串

func main() {

}

// 双指针
func reverseString(s []byte) {
	if s == nil || len(s) < 2 {
		return
	}

	l, r := 0, len(s)
	for l < r {
		s[l], s[r] = s[r], s[l]
		l++
		r--
	}
}

File: C++/344.反转字符串.cpp

// 344. 反转字符串

#include <algorithm>
#include <vector>

class Solution {
public:
    void reverseString(std::vector<char>& s) {
        for (int l = 0, r = s.size() - 1; l < r; ++l, --r) {
            std::swap(s[l], s[r]);
        }
    }
};
